src/software/thunderscope/thunderscope.py

```diff
@@ -47,46 +81,225 @@ class Thunderscope(object):
 
         self.window = QtGui.QMainWindow()
-        self.window.setCentralWidget(self.dock_area)
+        self.window.setCentralWidget(self.tabs)
         self.window.setWindowTitle("Thunderscope")
 
-        # Setup unix socket directory
-        try:
-            os.mkdir("/tmp/tbots")
-        except:
-            pass
+        # ProtoUnixIOs
+        #
+        # NOTE: We have two separate IOs for each full system because the
+        # er force simulator expects two inputs of the same protobuf type but
+        # from the blue or yellow team. We also would like to visualize the same
+        # protobuf types on two separate widgets.
+        self.simulator_proto_unix_io = (
+            ProtoUnixIO()
+            if simulator_proto_unix_io is None
+            else simulator_proto_unix_io
+        )
+        self.yellow_full_system_proto_unix_io = (
+            ProtoUnixIO()
+            if yellow_full_system_proto_unix_io is None
+            else yellow_full_system_proto_unix_io
+        )
+        self.blue_full_system_proto_unix_io = (
+            ProtoUnixIO()
+            if blue_full_system_proto_unix_io is None
+            else blue_full_system_proto_unix_io
+        )
+
+        self.refresh_timers = []
+
+        # Save and Load Prompts
+        self.save_layout_shortcut = QtGui.QShortcut(
+            QtGui.QKeySequence("Ctrl+S"), self.window
+        )
+        self.save_layout_shortcut.activated.connect(self.save_layout)
+
+        self.open_layout_shortcut = QtGui.QShortcut(
+            QtGui.QKeySequence("Ctrl+O"), self.window
+        )
+        self.open_layout_shortcut.activated.connect(self.load_layout)
+
+        self.reset_layout_shortcut = QtGui.QShortcut(
+            QtGui.QKeySequence("Ctrl+R"), self.window
+        )
+
+        def __reset_layout():
+            if os.path.exists(DEFAULT_LAYOUT_PATH):
+                os.remove(DEFAULT_LAYOUT_PATH)
+                QMessageBox.information(
+                    self.window,
+                    "Restart Required",
+                    """
+                    Restart thunderscope to reset the layout.
+                    """,
+                )
+
+        self.reset_layout_shortcut.activated.connect(__reset_layout)
+
+        self.show_help = QtGui.QShortcut(QtGui.QKeySequence("h"), self.window)
+        self.show_help.activated.connect(
+            lambda: QMessageBox.information(
+                self.window,
+                "Help",
+                """
+                Cntrl+S: Save Layout
+                Cntrl+O: Open Layout
+                Cntrl+R: Reset Layout
+                Double Click Purple Bar to pop window out
+                Drag Purple Bar to rearrange docks
+                Click items in legends to select/deselect
+
+                Cntrl-Click and Drag: Move ball and kick
+                Click and Drag Robots to move them around
+                I to identify robots, show their IDs
```

Comment:
is this the best we can do?

Reply:
always drawing the ID is very cluttered, it also seems to tear the geometry nearby

I think there is a way to paint the text on screen but I was thinking we don't need to see the robot ids

We might need to check the goalie id temporarily, but ideally our code should be id agnostic anyway so its not as useful to draw on screen all the time (IMO)

thoughts? cc @nimazareian @itsarune